azext_edge/edge/providers/rpsaas/adr2/assets.py

```diff
@@ -0,0 +1,1078 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+import json
+from typing import TYPE_CHECKING, Dict, List, Iterable, Optional, Tuple, Union
+from knack.log import get_logger
+from azure.cli.core.azclierror import (
+    InvalidArgumentValueError,
+    RequiredArgumentMissingError,
+)
+from azure.core.exceptions import ResourceNotFoundError
+
+from .user_strings import MISSING_DATA_EVENT_ERROR, ENDPOINT_NOT_FOUND_WARNING, INVALID_OBSERVABILITY_MODE_ERROR
+from ....util import assemble_nargs_to_dict
+from ....common import FileType
+from ....util.az_client import get_registry_mgmt_client
+from ....util.queryable import Queryable
+
+if TYPE_CHECKING:
+    from ....vendor.clients.deviceregistrymgmt.operations import (
+        AssetsOperations,
+        DiscoveredAssetsOperations
+    )
+
+logger = get_logger(__name__)
+
+ASSET_RESOURCE_TYPE = "Microsoft.DeviceRegistry/assets"
+DISCOVERED_ASSET_RESOURCE_TYPE = "Microsoft.DeviceRegistry/discoveredAssets"
+VALID_DATA_OBSERVABILITY_MODES = ["None", "Gauge", "Counter", "Histogram", "Log"]
+VALID_EVENT_OBSERVABILITY_MODES = ["None", "Log"]
+
+class Assets(Queryable):
+    def __init__(self, cmd):
+        super().__init__(cmd=cmd)
+        self.deviceregistry_mgmt_client = get_registry_mgmt_client(
+            subscription_id=self.default_subscription_id
+        )
+        self.ops: "AssetsOperations" = self.deviceregistry_mgmt_client.assets
+        self.discovered_ops: "DiscoveredAssetsOperations" = self.deviceregistry_mgmt_client.discovered_assets
```

Comment:
I can comment out everything with respect to discovered assets so it can be easily put back in. For now just use assets